src/tactics/FStar.Tactics.V2.Basic.fst

```diff
@@ -2385,14 +2445,18 @@ let refl_instantiate_implicits (g:env) (e:term) : tac (option (term & typ) & iss
     let must_tot = true in
     let g = {g with instantiate_imp=false; phase1=true; lax=true} in
     let e, t, guard = g.typeof_tot_or_gtot_term g e must_tot in
+    (* We force this guard here, and do not delay it, since we
+    will return this term and it MUST be compressed. We could
+    split the logical part away and return that, though. *)
```

Comment:
This is a lax call though, so dropping the logical payload of the guard is totally fine.

Reply:
Good point! I changed that comment slightly.